src/transforms/log_to_metric.rs

```diff
@@ -115,6 +116,72 @@ impl LogToMetric {
     }
 }
 
+fn render_template(s: &str, event: &Event) -> Result<String, String> {
+    let template = Template::from(s);
+    let name = template
+        .render(&event)
+        .map_err(|e| format!("Keys ({:?}) do not exist on the event. Dropping event.", e))?;
+    Ok(String::from_utf8_lossy(&name.to_vec()).to_string())
+}
+
+fn to_metric(config: &MetricConfig, event: &Event) -> Result<Metric, String> {
+    let log = event.as_log();
+    match config {
+        MetricConfig::Counter(counter) => {
+            let val = log.get(&counter.field).ok_or("")?;
```

Comment:
I think this is actually a change in behavior. In the previous version, only events where the counter field was set would result in a metric value being produced. Clearly we're missing a test that specifies this, but I think that's still the behavior we want.

It looks like we're following the same pattern for the other metric types too, so maybe it'd make sense to do some extraction of the lookup logic to make sure we're consistent.

Reply:
Hm.. I was trying to keep existing behaviour untouched, please check `fn missing_field` test.
The idea here is that ok_or("")? pattern will pass only found items, and thus metric won't be produced. Maybe I'm missing the point here?

Reply:
Oh, I completely misread that, I'm sorry! For some reason, I thought it was `unwrap_or`.

I will say that an empty string as an error is a little confusing. I know we're not doing much with it, but something like `log.get(&field).ok_or(Error::MissingField)` would be much clearer.

Reply:
That said, I don't want to block this change on doing a full error type implementation. Even just a small change like `.ok_or("missing field")` would be totally fine for now.